src/modules/sub-items/components/sub-items-list/sub.items.list.component.js

```diff
@@ -1,34 +1,21 @@
-import React, { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 
 import TableViewComponent from '../table-view/table.view.component.js';
 import GridViewComponent from '../grid-view/grid.view.component.js';
 
 import './css/sub.items.list.component.css';
 
-export default class SubItemsListComponent extends Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            activeView: props.activeView,
-            views: {
-                table: TableViewComponent,
-                grid: GridViewComponent,
-            },
-        };
-    }
-
-    UNSAFE_componentWillReceiveProps(props) {
-        this.setState((state) => ({ ...state, activeView: props.activeView }));
-    }
+const views = {
+    table: TableViewComponent,
+    grid: GridViewComponent,
```

Comment:
Maybe it would be a good idea to extract `table` and `grid` keys to external const variables, because we use them here and in other place - we get one of these keys from `props`.

e.g.:
```js
export const VIEW_KEY_TABLE = 'table';
```

Reply:
In `/src/modules/sub-items/components/view-switcher/view.switcher.component.js` in line 76 we have: 
```js
{this.renderButton('table', 'view-list', listViewBtnLabel)}
{this.renderButton('grid', 'view-grid', gridViewBtnLabel)}
```
It is good practice to extract common constants to one shared variable.

Reply:
We also have in `/src/modules/sub-items/sub.items.module.js` in line 372: `activeView: 'table',`

Reply:
In our current codebase it's not needed. I know, you're trying to follow someone's good practices but we have our own standards. In redux, there's a pattern to expose the action names as constants. That's fine. But we don't need it here, as it's not going to change in the lifecycle of this app. In the future, if there's a need to change it, then we'll change it. I'm not a big fan of premature optimizations.